core/templates/dev/head/pages/exploration_editor/simple_editor_tab/SimpleEditorManagerServiceSpec.js

```diff
@@ -376,6 +376,77 @@ describe('Simple Editor Manager Service', function() {
       })).toBe(true);
   });
 
+  describe('deleteQuestion', function () {
+    var stateNameOfQuestionToDelete;
+    var questionToDelete;
+    var nextStateName;
+
+    beforeEach(function(){
+      simpleEditorManagerService.tryToInit();
+      stateNameOfQuestionToDelete = 'Question 1';
+      nextStateName = 'Question 2';
+      expect(function(){simpleEditorManagerService.getQuestionList()
+        .getBindableQuestion(stateNameOfQuestionToDelete)}).not.toThrowError();
+      questionToDelete = questionList
+        .getBindableQuestion(stateNameOfQuestionToDelete);
+      expect(explorationStatesService.hasState(stateNameOfQuestionToDelete))
+        .toBe(true);
+    });
+
+    it('should remove question from question list', function () {
+      simpleEditorManagerService.deleteQuestion(questionToDelete);
+      expect(simpleEditorManagerService.getQuestionList().getQuestions()
+        .indexOf(questionToDelete)).toEqual(-1);
+    });
+
+    it('should remove the corresponding state', function(){
+      simpleEditorManagerService.deleteQuestion(questionToDelete);
+      expect(explorationStatesService.hasState(stateNameOfQuestionToDelete))
+        .toBe(false);
+    });
+
+    it('should change the destination of states that point to it to' +
+      'next state.', function(){
+      var stateThatPointToIt = 'Introduction';
+      simpleEditorManagerService.deleteQuestion(questionToDelete);
+      var expectedDestination = simpleEditorShimService
+          .getState(stateThatPointToIt).interaction
+            .answerGroups[0].outcome.dest;
+      expect(expectedDestination).toEqual(nextStateName);
+    });
+
+    it('should change the destination of questions that point to it to' +
+      'next question',function(){
+      var stateThatPointToIt = 'Introduction';
+      var questionThatPointToIt = questionList
+        .getBindableQuestion(stateThatPointToIt);
+      simpleEditorManagerService.deleteQuestion(questionToDelete);
+      var expectedDestination = simpleEditorManagerService.getQuestionList()
+        .getBindableQuestion(stateThatPointToIt)._answerGroups[0].outcome.dest;
```

Comment:
Perhaps we should add a getAnswerGroups() method or similar to the Question object, rather than reference a private member here?